datafusion/execution/src/object_store.rs

```diff
@@ -79,71 +79,42 @@ impl std::fmt::Display for ObjectStoreUrl {
     }
 }
 
-/// Provides a mechanism for lazy, on-demand creation of an [`ObjectStore`]
```

Comment:
We seem to have lost a significant amount of documentation -- maybe it was lost in a merge?

It would be great to restore the docstrings unless there is some reason to remove them

Reply:
My bad. I'll raise another commit to back the comments with examples.